src/main/java/codeu/controller/RegisterServlet.java

```diff
@@ -35,30 +40,57 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
      throws IOException, ServletException {
 
      String username = request.getParameter("username");
-     String password = request.getParameter("password");
+     String email = request.getParameter("email");     
+	 String password = request.getParameter("password");
      String passwordHash = BCrypt.hashpw(password, BCrypt.gensalt());
 
    //alphanumeric check
    if (!username.matches("[\\w*\\s*]*")) {
-   		request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
+   		request.setAttribute("error", "Invalid username. Please enter only letters, numbers, and spaces.");
      	request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
      	return;
    }
-
+	// if username is empty
+   if (username.matches("")){

```

Comment:
Can you add a space between the last ')' and '{' ?

Reply:
Is this mostly for documentation or could it cause any issues?

Reply:
just style!